### PR TITLE
feat(route53): added EvaluateTargetHealth to Route53 Alias targets (#9481)

### PR DESCRIPTION
**Pull request by wladyslawczyzewski**
### Issue # (if applicable)

Part of a bigger piece of work for #9481 
Another PR for this ticket #30739 

### Reason for this change

At this moment constructing the Route53 health checks is available only via L1 construct, this PR is one of the series to introduce the R53 health checks via higher-level constructs.

### Description of changes

The underlying L1 construct of a few L2 constructs for Alias targets already has the `EvaluateTargetHealth` property, so by adding this prop to the L2 construct we get the quick win to introduce the health checks for Alias records.

### Description of how you validated changes

- unit tests
- integration tests

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by wladyslawczyzewski** [comment]
hey guys, this is my first contribution to this repo, I would be grateful for any feedback 🤝

File: packages/aws-cdk-lib/aws-route53/lib/alias-record-target.ts
Comment (badmintoncryer) [inline packages/aws-cdk-lib/aws-route53/lib/alias-record-target.ts:32]: The default value is `undefined`, but could you describe what settings are applied when it is `undefined`?

File: packages/aws-cdk-lib/aws-route53-targets/lib/bucket-website-target.ts
Comment (badmintoncryer) [inline packages/aws-cdk-lib/aws-route53-targets/lib/bucket-website-target.ts:39]:
```suggestion
   * Evaluate target health
   *
   * @default - no health check configuration
```

File: packages/aws-cdk-lib/aws-route53-targets/lib/classic-load-balancer-target.ts
Comment (badmintoncryer) [inline packages/aws-cdk-lib/aws-route53-targets/lib/classic-load-balancer-target.ts:25]:
```suggestion
   * Evaluate target health
   *
   * @default - no health check configuration
```

File: packages/aws-cdk-lib/aws-route53-targets/lib/elastic-beanstalk-environment-target.ts
Comment (badmintoncryer) [inline packages/aws-cdk-lib/aws-route53-targets/lib/elastic-beanstalk-environment-target.ts:47]:
```suggestion
   * Evaluate target health
   *
   * @default - false
```

File: packages/aws-cdk-lib/aws-route53-targets/lib/global-accelerator-target.ts
Comment (badmintoncryer) [inline packages/aws-cdk-lib/aws-route53-targets/lib/global-accelerator-target.ts:47]:
```suggestion
   * Evaluate target health
   *
   * @default - no health check configuration
```

File: packages/aws-cdk-lib/aws-route53-targets/lib/load-balancer-target.ts
Comment (badmintoncryer) [inline packages/aws-cdk-lib/aws-route53-targets/lib/load-balancer-target.ts:25]:
```suggestion
   * Evaluate target health
   *
   * @default - no health check configuration
```

File: packages/aws-cdk-lib/aws-route53-targets/lib/elastic-beanstalk-environment-target.ts
Comment (badmintoncryer) [inline packages/aws-cdk-lib/aws-route53-targets/lib/elastic-beanstalk-environment-target.ts:15]: Could you explain why only the target for Beanstalk has false specified as the default value?

**Review by badmintoncryer — CHANGES_REQUESTED** [review]
Thank you for your contribution! 

Could you also create integ tests for targets other than ALB and Global Accelerator?

File: packages/aws-cdk-lib/aws-route53-targets/lib/elastic-beanstalk-environment-target.ts
Comment (wladyslawczyzewski) [reply packages/aws-cdk-lib/aws-route53-targets/lib/elastic-beanstalk-environment-target.ts]: hi @badmintoncryer , thanks for your review. This change should be reverted in one of previous commits but looks like I missed that. Thanks for catching this!

File: packages/aws-cdk-lib/aws-route53/lib/alias-record-target.ts
Comment (wladyslawczyzewski) [reply packages/aws-cdk-lib/aws-route53/lib/alias-record-target.ts]: hi @badmintoncryer , I've changed this to more descriptive information about the default value of this property. Could you please check and let me know what do you think?

**Comment by wladyslawczyzewski** [comment]
hi @badmintoncryer thanks fro your review. I have applied the changes you suggested and added few more integration test cases. Could you please check these changes and let me know your thoughts?

**Comment by badmintoncryer** [comment]
@wladyslawczyzewski Thank you for your modification. I'll check and approve this PR later. Please wait for a while

**Comment by badmintoncryer** [comment]
@wladyslawczyzewski Could you please add integ test for `ElasticBeanstalkEnvironmentEndpointTarget`? I plan to approve as soon as it’s added!

**Comment by wladyslawczyzewski** [comment]
hi @badmintoncryer I just added the test case for Elastic Beanstalk Environment target, please check it out!

**Review by badmintoncryer — APPROVED** [review]
Thanks!!

**Review by GavinZZ — DISMISSED** [review]
One comment on extracting the interfaces into a common interface.

File: packages/aws-cdk-lib/aws-route53-targets/lib/global-accelerator-target.ts
Comment (wladyslawczyzewski) [reply packages/aws-cdk-lib/aws-route53-targets/lib/global-accelerator-target.ts]: hi @GavinZZ , thanks for your review! I just extracted this common interface - could you please review the PR one more time?

File: packages/aws-cdk-lib/aws-route53-targets/lib/props.ts
Comment (GavinZZ) [inline packages/aws-cdk-lib/aws-route53-targets/lib/props.ts:11]: nit: Can you name the file `shared.ts` instead of `props`.

**Review by GavinZZ — CHANGES_REQUESTED** [review]
One more nit and I'm happy to approve.

File: packages/aws-cdk-lib/aws-route53-targets/lib/props.ts
Comment (wladyslawczyzewski) [reply packages/aws-cdk-lib/aws-route53-targets/lib/props.ts]: hi @GavinZZ just renamed the file, please check it out

**Review by GavinZZ — DISMISSED** [review]

**Review by GavinZZ — APPROVED** [review]

**Comment by GavinZZ** [comment]
@mergify update

**Comment by GavinZZ** [comment]
@wladyslawczyzewski can you fix the merge conflicts? I've approved the PR and will prioritize on merging this once you fixed the conflicts. Thank you!

**Review by GavinZZ — DISMISSED** [review]

**Comment by badmintoncryer** [comment]
@wladyslawczyzewski 
It’s better to comment `@mergify update` instead of manually merging the main branch after the maintainer’s approval. This allows you to update the branch without dismissing the maintainer’s review. 😁

**Review by kaizencc — APPROVED** [review]
Approving simply because this PR was previously approved by a maintainer

**Pull request merged**
